madlibs.js

```diff
@@ -24,12 +24,75 @@
  *  ....
  * 
  * There are multiple ways to do this, but you may want to use regular expressions.
- * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/
+ * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/ */
+
+
+
+/**
+ * A function that retrieves the story from story.txt and return an array of objects
+ * @param {String}  rawStory: string 
+ * @returns {Array<Objetc>} wordsObjetct: Array of objetcs
  */
 function parseStory(rawStory) {
-  // Your code here.
-  return {}; // This line is currently wrong :)
+  /* *
+   * An Object to maps part of speech codes to their corresponding labels
+   */
+  const partOfSpeech = {
+    n: "[noun]",
+    v: "[verb]",
+    a: "[adjectif]",
+    ad: "[adverb]",
+
+  }
+
+  /**
+   * Regex matches words with part of speech codes (e.g., "word[pos]") 
+   * Or just words
+   * 
+   */
+  const matchWordsRegex = /\w+\[[anvad]+\]|[.,]|\w+/g
+
+
+  /**
+   * Regex to match the format "word[pos]" 
+   * And captures both the word and the part of speech code.
+   */
+  const matchPosRegex = /(\w+)\[([anvad]+)\]/
```

Comment:
It's great that you considered adding another Regex instead of a switch case. Thinking outside the box!